big-red-shuttle/ScheduleBar.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  ScheduleBar.swift
+//  big-red-shuttle
+//
+//  Created by Monica Ong on 11/3/16.
+//  Copyright © 2016 cuappdev. All rights reserved.
+//
+
+import UIKit
+
+protocol ScheduleBarDelegate {
+    func selectCell(button: UIButton) -> Void
+}
+
+class ScheduleBar: UIScrollView {
+    private var buttons: [UIButton]!
+    private var oval: UIView!
+    private var seperator: UIView!
+    private var buttonWidth: CGFloat!
+    private var buttonPadding: CGFloat = 20
+    private var selectedButton: UIButton!
+    var delegateSB: ScheduleBarDelegate!
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setUp(buttonsData: [String], selected: Int){
+        self.backgroundColor = UIColor.white
+        
+        buttons = []
+        for time in buttonsData{ //set up buttons
+            let button = UIButton(frame: CGRect(x: 0, y: 0, width: 50, height: self.bounds.height))
+            button.setTitle(time, for: UIControlState.normal)
+            button.setTitleColor(Color.greyedout, for: UIControlState.normal)
+            button.titleLabel?.font = UIFont(name: "HelveticaNeue-Medium", size: 13.0)
+            button.sizeToFit()
+            //Make oval background view behind button
+            var frameRect: CGRect = button.frame
+            buttonWidth = button.bounds.width + buttonPadding
+            frameRect.size.width = buttonWidth
+            button.frame = frameRect
```

Comment:
combine lines 43 - 46 to `button.frame.size.width = button.bounds.width + buttonPadding`